packages/@react-stately/steplist/src/useStepListState.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {AriaStepListProps, StepListState} from '@react-types/steplist';
+import {Collection, Node} from '@react-types/shared';
+import {Key, useEffect, useMemo} from 'react';
+import {useControlledState} from '@react-stately/utils';
+import {useSingleSelectListState} from '@react-stately/list';
+
+export function useStepListState<T extends object>(props: AriaStepListProps<T>): StepListState<T> {
+  let state = useSingleSelectListState<T>(props);
+
+  let [lastCompletedStep, setLastCompletedStep] = useControlledState<Key>(props.lastCompletedStep, props.defaultLastCompletedStep, props.onLastCompletedStepChange);
+  const {setSelectedKey: realSetSelectedKey, selectedKey, collection} = state;
+  const {prevKeyMap, indexMap} = useMemo(() => buildKeysMaps(collection), [collection]);
+  const disabledKeys = useMemo(() => new Set(props.disabledKeys), [props.disabledKeys]);
```

Comment:
not sure if `disabledKeys` should fallow same `approach` as `tabs` `breadcrumbs` ... it feels weird to me that if step `3` is disabled user can still navigate to `4-5-6....` since i feel steps should be completed in `sequence`, i fallowed up with `design` on this one. If someone has more info on this pls feel free to share.